middleware/verify_privilege_token/main.go

```diff
@@ -0,0 +1,64 @@
+package verify_privilege_token
+
+import (
+	"fmt"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/rs/zerolog"
+	"golang.org/x/exp/slices"
+)
+
+type VerifyPrivilegeTokenInterface interface {
+	HasTokenPrivilege(privilegeID int64) fiber.Handler
+}
+
+type Config struct {
+	Log *zerolog.Logger
+}
+
+type verifyPrivilegeToken struct {
+	cfg Config
+}
+
+func New(cfg Config) VerifyPrivilegeTokenInterface {
+	return &verifyPrivilegeToken{
+		cfg: cfg,
+	}
+}
+
+func (p *verifyPrivilegeToken) HasTokenPrivilege(privilegeID int64) fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		return p.checkPrivilege(c, privilegeID)
+	}
+}
+
+func (p *verifyPrivilegeToken) checkPrivilege(c *fiber.Ctx, privilegeID int64) error {
+	logger := p.cfg.Log
```

Comment:
We could log [the route](https://docs.gofiber.io/api/ctx#route) in this method. It's always a bit tricky to log in libraries, you could say that this isn't our job here. You can make a sub-logger with a fixed attribute by doing something like

```
logger := p.cfg.Log.With().Str("myAttr", "xdd").Logger()
```

Note that `logger` here is not a pointer for some reason. We might also log the user once we have it, just so we know who's having a bad time.